libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/UnitTestTools.hs

```diff
@@ -0,0 +1,184 @@
+  rewardCoin,
+  depositCoin,
+
+  -- * Types supported
+  Hash,
+  Addr (..),
+  RewardAccount (..),
+  Coin (..),
+  CompactForm (CompactCoin),
+  Credential (..),
+  StakeReference (..),
+  StandardCrypto,
+  Crypto,
+  DRep (DRepCredential),
+  KeyHash,
+  KeyRole (..),
+  SafeHash,
+  TxId,
+  TxIn,
+  IndividualPoolStake (IndividualPoolStake),
+  WitVKey,
+  RDPair (..),
+  UTxO (..),
+  PoolParams (..),
+  CertState (..),
+  DState (..),
+  PState (..),
+  VState (..),
+  StrictMaybe (..),
+) where
+
+import Cardano.Crypto.DSIGN.Class ()
+import Cardano.Ledger.Address (Addr (..), RewardAccount (..))
+import Cardano.Ledger.BaseTypes (Network (..), StrictMaybe (..))
+import Cardano.Ledger.CertState (CertState (..), DState (..), PState (..), VState (..))
+import Cardano.Ledger.Coin (Coin (..), CompactForm (CompactCoin))
+import Cardano.Ledger.Compactible (fromCompact)
+import Cardano.Ledger.Core
+import Cardano.Ledger.Credential (Credential (..), StakeReference (..))
+import Cardano.Ledger.Crypto (Crypto, StandardCrypto)
+import Cardano.Ledger.DRep (DRep (DRepCredential))
+import Cardano.Ledger.Keys (Hash, KeyHash (..), KeyRole (..), WitVKey (..), coerceKeyRole, hashKey)
+import Cardano.Ledger.PoolDistr (IndividualPoolStake (IndividualPoolStake))
+import Cardano.Ledger.PoolParams (PoolParams (..))
+import Cardano.Ledger.SafeHash (SafeHash)
+import Cardano.Ledger.TxIn (TxId (..), TxIn (..))
+import Cardano.Ledger.UMap (RDPair (..))
+import Cardano.Ledger.UTxO (UTxO (..))
+import Cardano.Ledger.Val (Val (..))
+import Data.Default.Class (Default (def))
+import Data.Kind (Type)
+import GHC.TypeNats (KnownNat, Nat)
+import Lens.Micro
+import Test.Cardano.Ledger.Core.Arbitrary ()
+import Test.Cardano.Ledger.Core.KeyPair (KeyPair (..), mkWitnessVKey)
+import Test.QuickCheck
+
+-- ==================================================
+
+-- | Supports patterns like
+--   (tom, john, ann, ron, mary) <- fresh @5 @(Credential 'Staking (EraCrypto era))
+class KnownNat n => Fresh (n :: Nat) (a :: Type) where
+  type Tuple n a :: Type
+  fresh :: Arbitrary a => Tuple n a
+
+instance Fresh 1 a where
+  type Tuple 1 a = Gen a
+  fresh = arbitrary @a
+
+instance Fresh 2 a where
+  type Tuple 2 a = Gen (a, a)
+  fresh = (arbitrary @(a, a))
+
+instance Fresh 3 a where
+  type Tuple 3 a = Gen (a, a, a)
+  fresh = (arbitrary @(a, a, a))
+
+instance Fresh 4 a where
+  type Tuple 4 a = Gen (a, a, a, a)
+  fresh = (arbitrary @(a, a, a, a))
+
+instance Fresh 5 a where
+  type Tuple 5 a = Gen (a, a, a, a, a)
+  fresh = (arbitrary @(a, a, a, a, a))
+
+instance Fresh 6 a where
+  type Tuple 6 a = Gen (a, a, a, a, a, a)
+  fresh = (arbitrary @(a, a, a, a, a, a))
+
+instance Fresh 7 a where
+  type Tuple 7 a = Gen (a, a, a, a, a, a, a)
+  fresh = (arbitrary @(a, a, a, a, a, a, a))
+
+genLens :: Gen a -> Lens' a b -> b -> Gen a
+genLens gen l b = do a <- gen; pure (a & l .~ b)
+
+arbitraryLens :: Arbitrary a => Lens' a b -> b -> Gen a
+arbitraryLens = genLens arbitrary
```

Comment:
These helpers aren't very useful, see my other comment how `set` can be used to get the desired behavior in much more clear fashion
```suggestion
```

Reply:
see above.